src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -824,19 +821,18 @@ impl Coordinator {
             return Err(AdapterError::BadItemInStorageCluster { cluster_name });
         }
 
-        // Generate the optimized expression, which allows us to know exacly what we depend on.
-        let optimized_expr = self.view_optimizer.optimize(view_expr)?;
-        let depends_on: Vec<_> = optimized_expr.depends_on().into_iter().collect();
-
-        // Validate after generating the optimized expression.
-        self.validate_timeline_context(depends_on.clone())?;
-        self.validate_system_column_references(ambiguous_columns, &depends_on)?;
-
+        // Validate any references in the materialized view's expression. We do
+        // this on the unoptimized plan to better reflect what the user typed.
+        // We want to reject queries that depend on log sources, for example,
+        // even if we can *technically* optimize that reference away.
```

Comment:
Do we want to add this same comment to `generate_view_ops`?

Reply:
Good idea!

Reply:
Done.